test/with_dummyserver/test_connectionpool.py

```diff
@@ -281,6 +282,22 @@ def test_unicode_upload(self) -> None:
             r = pool.request("POST", "/upload", fields=fields)
             assert r.status == 200, r.data
 
+    def test_streaming_multipart_upload(self) -> None:
+        body = MultipartEncoder(
+            {
+                "upload_flag": "enabled",
+                "upload_key": "value",
+                "upload_file": (
+                    "filename.txt",
+                    open(__file__, "rb"),
+                    {"Content-Type": "text/plain"},
+                ),
+            }
+        )
+        with HTTPConnectionPool(self.host, self.port) as pool:
+            r = pool.request("POST", "/upload", headers=body.request_headers, body=body)
```

Comment:
I suspect this will fail mypy but I thought it'd be good to have an end-to-end test here. It's also failing against the dummyserver for some reason I haven't figured out quite yet, but I'm working on that